shiny/ui/_markdown_stream.py

```diff
@@ -0,0 +1,224 @@
+
+        This method is only relevant for Shiny Express. In Shiny Core, use
+        :func:`~shiny.ui.output_markdown_stream` for placing the markdown stream
+        in the UI.
+
+        Returns
+        -------
+        Tag
+            The UI element for this markdown stream.
+        """
+        return output_markdown_stream(self.id, self._content, self._content_type)
+
+    def stream(self, content: Iterable[str], clear: bool = True):
+        """
+        Stream content into the markdown stream.
+
+        Parameters
+        ----------
+        content
+            The content to stream. This can be any iterable of strings, such as a list,
+            generator, or file-like object.
+        clear
+            Whether to clear the existing content before streaming the new content.
+        """
+
+        @reactive.extended_task
+        async def _task():
+            if clear:
+                self._replace("")
+            with self._streaming_dot():
+                for c in content:
+                    self._append(c)
+
+        _task()
+
+        # Since the task runs in the background (outside/beyond the current context,
+        # if any), we need to manually raise any exceptions that occur
+        @reactive.effect
+        async def _handle_error():
+            e = _task.error()
+            if e:
+                await self._raise_exception(e)
+            _handle_error.destroy()  # type: ignore
+
+    def _append(self, content: str):
+        msg: ContentMessage = {
+            "id": self.id,
+            "content": content,
+            "operation": "append",
+        }
+
+        self._send_custom_message(msg)
+
+    def _replace(self, content: str):
+        msg: ContentMessage = {
+            "id": self.id,
+            "content": content,
+            "operation": "replace",
+        }
+
+        self._send_custom_message(msg)
+
+    @contextmanager
+    def _streaming_dot(self):
+        start: isStreamingMessage = {
+            "id": self.id,
+            "isStreaming": True,
+        }
+        self._send_custom_message(start)
+
+        try:
+            yield
+        finally:
+            end: isStreamingMessage = {
+                "id": self.id,
+                "isStreaming": False,
+            }
+            self._send_custom_message(end)
+
+    async def _raise_exception(self, e: BaseException):
+        if self.on_error == "unhandled":
+            raise e
+        else:
+            sanitize = self.on_error == "sanitize"
+            msg = f"Error in MarkdownStream('{self.id}'): {str(e)}"
+            raise NotifyException(msg, sanitize=sanitize) from e
+
+    def _send_custom_message(self, msg: Union[ContentMessage, isStreamingMessage]):
+        if self._session.is_stub_session():
+            return
+        self._session._send_message_sync(
+            {"custom": {"shinyMarkdownStreamMessage": msg}}
+        )
+
+
+@add_example()
+def output_markdown_stream(
```

Comment:
FWIW, I feel a bit torn on whether this should be `output_markdown_stream()` or `markdown_stream_ui()`

Reply:
[3b4a5a4](https://github.com/posit-dev/py-shiny/pull/1782/commits/3b4a5a4f7e70ffcd028a5f8229779496bc66ff2b) does this renaming

Reply:
lol I called this out in comments, I think the paired `@render...` function detail is probably not how most users will think of what's an output vs what's an input. I think that most people will see this as an "output" because you're sending something from the server and it's showing up in the UI.